src/language-html/ast.js

```diff
@@ -96,54 +98,71 @@ class Node {
    * @param {Object} [node]
    */
   insertChildBefore(target, node) {
-    // @ts-expect-error
-    this.children.splice(
-      // @ts-expect-error
-      this.children.indexOf(target),
-      0,
-      this.createChild(node),
-    );
+    const children = this.$children;
+    children.splice(children.indexOf(target), 0, this.createChild(node));
   }
 
   /**
    * @param {Node} [child]
    */
   removeChild(child) {
-    // @ts-expect-error
-    this.children.splice(this.children.indexOf(child), 1);
+    const children = this.$children;
+    children.splice(children.indexOf(child), 1);
   }
 
   /**
    * @param {Node} [target]
    * @param {Object} [node]
    */
   replaceChild(target, node) {
-    // @ts-expect-error
-    this.children[this.children.indexOf(target)] = this.createChild(node);
+    const children = this.$children;
+    children[children.indexOf(target)] = this.createChild(node);
   }
 
   clone() {
     return new Node(this);
   }
 
+  get #childrenProperty() {
+    if (this.type === "angularIcuCase") {
+      return "expression";
+    }
+
+    if (this.type === "angularIcuExpression") {
+      return "cases";
+    }
+
+    return "children";
+  }
+
+  get $children() {
+    return this[this.#childrenProperty];
+  }
+
+  set $children(value) {
+    this[this.#childrenProperty] = value;
+  }
```

Comment:
What does the `$` prefix mean?

Reply:
`children` already used, can't use `#children`, do you have a better idea?

Reply:
What is the difference from the `children` property already in use? Please name them using that difference